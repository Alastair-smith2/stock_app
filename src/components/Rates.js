import React from "react";
import { View, Text, StyleSheet } from "react-native";

const Rates = ({ data }) => {
  console.log(data, "What is the data?");
  return (
    <View style={styles.container}>
      <View style={styles.rate}>
        <Text style={styles.text}>From - {data["2. From_Currency Name"]}</Text>
      </View>
      <View style={styles.padding} />
      <View style={styles.rate}>
        <Text style={styles.text}>To - {data["4. To_Currency Name"]}</Text>
      </View>
      <View style={styles.padding} />
      <View style={styles.rate}>
        <Text style={styles.text}>Rate - {data["5. Exchange Rate"]}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginBottom: 10,
    justifyContent: "center",
    alignItems: "center"
  },
  rate: {
    flex: 1
  },
  padding: {
    flex: 0.1
  }
});
export default Rates;
