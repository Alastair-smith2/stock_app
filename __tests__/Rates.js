import React from "react";
import { shallow } from "enzyme";

import Rates from "../src/components/Rates.js";

describe("The Rates component", () => {
  let wrapper;
  let mockData = {
    "2. From_Currency Name": "GBP",
    "4. To_Currency Name": "USD",
    "5. Exchange Rate": 2
  };
  console.log(mockData);
  beforeEach(() => {
    wrapper = shallow(<Rates data={mockData} />);
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
