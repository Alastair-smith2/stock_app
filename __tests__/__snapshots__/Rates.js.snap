// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Rates component It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rates
    data={
      Object {
        "2. From_Currency Name": "GBP",
        "4. To_Currency Name": "USD",
        "5. Exchange Rate": 2,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}
          >
            From - 
            GBP
          </Text>
        </View>,
        <View
          style={
            Object {
              "flex": 0.1,
            }
          }
        />,
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}
          >
            To - 
            USD
          </Text>
        </View>,
        <View
          style={
            Object {
              "flex": 0.1,
            }
          }
        />,
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}
          >
            Rate - 
            2
          </Text>
        </View>,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}
          >
            From - 
            GBP
          </Text>,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "From - ",
              "GBP",
            ],
            "ellipsizeMode": "tail",
            "style": undefined,
          },
          "ref": null,
          "rendered": Array [
            "From - ",
            "GBP",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "flex": 0.1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}
          >
            To - 
            USD
          </Text>,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "To - ",
              "USD",
            ],
            "ellipsizeMode": "tail",
            "style": undefined,
          },
          "ref": null,
          "rendered": Array [
            "To - ",
            "USD",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "flex": 0.1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}
          >
            Rate - 
            2
          </Text>,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "Rate - ",
              2,
            ],
            "ellipsizeMode": "tail",
            "style": undefined,
          },
          "ref": null,
          "rendered": Array [
            "Rate - ",
            2,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
              From - 
              GBP
            </Text>
          </View>,
          <View
            style={
              Object {
                "flex": 0.1,
              }
            }
          />,
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
              To - 
              USD
            </Text>
          </View>,
          <View
            style={
              Object {
                "flex": 0.1,
              }
            }
          />,
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
              Rate - 
              2
            </Text>
          </View>,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginBottom": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
              From - 
              GBP
            </Text>,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "From - ",
                "GBP",
              ],
              "ellipsizeMode": "tail",
              "style": undefined,
            },
            "ref": null,
            "rendered": Array [
              "From - ",
              "GBP",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "flex": 0.1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
              To - 
              USD
            </Text>,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "To - ",
                "USD",
              ],
              "ellipsizeMode": "tail",
              "style": undefined,
            },
            "ref": null,
            "rendered": Array [
              "To - ",
              "USD",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "flex": 0.1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
              Rate - 
              2
            </Text>,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "Rate - ",
                2,
              ],
              "ellipsizeMode": "tail",
              "style": undefined,
            },
            "ref": null,
            "rendered": Array [
              "Rate - ",
              2,
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
